pkg/apis/networking/validation/validation_test.go

```diff
@@ -1776,3 +2038,182 @@ func TestValidateIngressStatusUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateResourceBackendPresent(t *testing.T) {
+	implementationPathType := networking.PathTypeImplementationSpecific
+	defaultBackend := networking.IngressBackend{
+		ServiceName: "default-backend",
+		ServicePort: intstr.FromInt(80),
+	}
+	resourceBackend := &api.TypedLocalObjectReference{
+		APIGroup: utilpointer.StringPtr("example.com"),
+		Kind:     "foo",
+		Name:     "bar",
+	}
+	baseIngress := networking.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Spec: networking.IngressSpec{
+			Backend: &networking.IngressBackend{
+				ServiceName: "default-backend",
+				ServicePort: intstr.FromInt(80),
+			},
+			Rules: []networking.IngressRule{
+				{
+					Host: "foo.bar.com",
+					IngressRuleValue: networking.IngressRuleValue{
+						HTTP: &networking.HTTPIngressRuleValue{
+							Paths: []networking.HTTPIngressPath{
+								{
+									Path:     "/foo",
+									PathType: &implementationPathType,
+									Backend:  defaultBackend,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Status: networking.IngressStatus{
+			LoadBalancer: api.LoadBalancerStatus{
+				Ingress: []api.LoadBalancerIngress{
+					{IP: "127.0.0.1"},
+				},
+			},
+		},
+	}
+	testCases := map[string]struct {
+		groupVersion         *schema.GroupVersion
+		tweakIngress         func(ing *networking.Ingress)
+		allowResourceBackend bool
+		expectedErrsOnFields []string
+	}{
+		"nil spec.Backend and no paths": {
+			tweakIngress: func(ing *networking.Ingress) {
+				ing.Spec.Backend = &networking.IngressBackend{}
```

Comment:
the description says `nil spec.Backend`, but this isn't nil

Reply:
spoke offline... many of the tests in this test method are setting non-nil fields to pass validation, but I was actually envisioning a low-level unit test that called resourceBackendPresent directly and verified the boolean result... no validation error checks needed, just wanted to make sure we didn't NPE ever and correctly detected all the scenarios where a resource backend was present